docs/reference/utilities.rst

```diff
@@ -21,11 +21,7 @@ Summary Classes
 .. autosummary::
    :toctree: api/
 
-   XRateLimitRule
    SleepingRateLimitRule
-   RateLimitRule
    RateLimiter
    DefaultRateLimiter
    RequestRate
-   LimitStructure
-   LimitsStructure
```

Comment:
I'm not sure if these are the only changes I need to make to the docs. Building the docs locally still fails with errors caused by these classes not existing anymore.

Reply:
i can have a look at the docs if you'd like. sorry i've been so quiet. the past few weeks have been really busy w work travel and then catchup when i got back. i'll try to check this out this weekend / early next week.

Reply:
ok the docs was a quick check as i suspected a simple solution. when you build the docs the API docs are build on top of stub files that are automagically generated by sphinx when we build. These files live in the docs/reference/api dir. When we remove methods, classes etc we can do 1 of two things. 

1. we can clean the entire API directory (probably best) and let sphinx autogenerate new API stubs for rendering
2. we can just remove the files related to removed methods, etc

so below that XRateLimitRule.rst file is still in our API stubs folder which throws an warning because it's a file that represents a docstring that doesn't exist because it was removed. 

https://github.com/stravalib/stravalib/blob/main/docs/reference/api/stravalib.util.limiter.XRateLimitRule.rst

Locally i just cleaned out the entire directory and let it rebuild and the warnings went away. 

i can push a fix if you are ok with that. It will add a bunch of modified/removed rst files in this pr however. are you ok with my doing that? or do you prefer to do it?

also in a future fix if we wanted i could add a step to our [nox clean_docs](https://github.com/stravalib/stravalib/blob/main/noxfile.py#L86) function to clean that directory as well? we'd want to remember to rebuild is the only thing and push the stubs up after

Reply:
on another note i just asked a colleague whether these stubs need to actually be in version control at all. maybe we can just remove them.

Reply:
Thanks for your other PR #447, that solved the issue!